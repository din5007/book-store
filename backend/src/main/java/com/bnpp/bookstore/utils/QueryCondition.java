package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
