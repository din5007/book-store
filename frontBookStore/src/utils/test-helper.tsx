import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;