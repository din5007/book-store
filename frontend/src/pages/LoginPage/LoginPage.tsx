
import React, { FC, useState } from "react";
import Button from "react-bootstrap/Button";
import Col from "react-bootstrap/esm/Col";
import Row from "react-bootstrap/esm/Row";
import Form from "react-bootstrap/Form";
import MainPage from "../MainPage/MainPage";
import { Link } from "react-router-dom";
import ErrorMessage from "../CommonPages/ErrorPage";
import Loading from "../CommonPages/LoadingPage";
import { useStoreActions, useStoreState } from "../../store/hooks";
import "./LoginPage.css";
import { useNavigate } from "react-router-dom";

const LoginPage : FC<{}> = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(false);
  const [loading, setLoading] = useState(false);
  const { logIn } = useStoreActions((actions) => actions.user);
  const navigate = useNavigate();
  

  const submitHandler = async (e: any) => {
    e.preventDefault();
    setLoading(true);
    try {
      logIn({ email, password });
      navigate('/books');
      setLoading(false);
    } catch (error : any) {
      setError(error.reponse.date.message);
      setLoading(false);
    }
  };

  return (
    <MainPage title="LOGIN">
      <div className="loginContainer">
        {error && <ErrorMessage variant="danger">{error}</ErrorMessage>}
        {loading && <Loading />}
        <Form onSubmit={submitHandler}>
          <Form.Group className="mb-3" controlId="formBasicEmail">
            <Form.Label>Email Address</Form.Label>
            <Form.Control
              type="email"
              value={email}
              placeholder="Enter Your Email Address"
              onChange={(e) => setEmail(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="formBasicPassword">
            <Form.Label>Password</Form.Label>
            <Form.Control
              type="password"
              value={password}
              placeholder="Enter Your Password"
              onChange={(e) => setPassword(e.target.value)}
            />
          </Form.Group>

          <Button variant="primary" type="submit">
            Submit
          </Button>
        </Form>
        <Row className="py-3">
          <Col>
            New User ? <Link to="/register">Register Here</Link>
          </Col>
        </Row>
      </div>
    </MainPage>
  );
};

export default LoginPage;
