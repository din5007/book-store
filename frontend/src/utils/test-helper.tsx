import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;