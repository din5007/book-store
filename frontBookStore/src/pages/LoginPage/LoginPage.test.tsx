import { customRender as render } from '../../utils/test-helper';
import LoginPage from "./LoginPage"
import { describe, expect, it } from 'vitest';

describe('Login Page', () => {
    it('Login Page should load properly - snapshot', async () => {
        const {container}  = render(<LoginPage />);
        expect(container).toMatchSnapshot();
    });
})