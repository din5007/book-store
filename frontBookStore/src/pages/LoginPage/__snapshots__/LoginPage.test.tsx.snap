// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > Login Page should load properly - snapshot 1`] = `
<div>
  <div
    style="min-height: 93vh; display: flex; padding: 10px 0px;"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="w-100"
        >
          <h1>
            LOGIN
          </h1>
          <div
            style="display: flex; flex-direction: column; margin: 20px;"
          >
            <form
              class=""
            >
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="formBasicEmail"
                >
                  Email Address
                </label>
                <input
                  class="form-control"
                  id="formBasicEmail"
                  placeholder="Enter Your Email Address"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="formBasicPassword"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  id="formBasicPassword"
                  placeholder="Enter Your Password"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
            <div
              class="py-3 row"
            >
              <div
                class="col"
              >
                New User ? 
                <a
                  href="/register"
                >
                  Register Here
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
